### DISCUSSION TITLE
Opening skin editor with any default skin breaks the game

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#22335] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

After opening skin editor game completely breaks that you can close osu! using ONLY the task manager.

### Screenshots or videos

https://user-images.githubusercontent.com/86934170/213888691-d5b1d117-aa1e-40a4-8c43-f11f01ad138b.mp4

### Version

121

### Logs

[network.log](https://github.com/ppy/osu/files/10473160/network.log)
[performance.log](https://github.com/ppy/osu/files/10473161/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10473162/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10473163/updater.log)
[database.log](https://github.com/ppy/osu/files/10473164/database.log)

**Comment by Loreos7** [comment]
Only happens in the gameplay. Doing the same in the song select only gives a small stutter. 

https://user-images.githubusercontent.com/86934170/213888803-b3d9084a-de81-43c7-bc43-8c313954697b.mp4

**Comment by bdach** [comment]
I have no idea why the logs are cut off but I have a feeling this is 99.9% duplicate of https://github.com/ppy/osu/issues/22326.

Anything in event viewer? Probably gonna be an access violation exception if I'm right.

**Comment by Loreos7** [comment]
They're cut off because game just breaks. The video shows that very clear.
It's not actual crush to desktop. It's just game-breaker thing. Game stops receiving any input after opening skin editor.

**Comment by bdach** [comment]
Yes, and now please check event viewer as instructed.

**Comment by Loreos7** [comment]
@bdach there is nothing in event viewer at all.

**Comment by bdach** [comment]
Are you able to test https://github.com/ppy/osu/pull/22330 and see if it fixes your issue?

**Comment by Loreos7** [comment]
@bdach nope, linked PR doesn't fix my problem.
Please note that this bug comes only with **default skins**. Custom skins or created ones from default [with "(modified)" brackets] works fine.

https://user-images.githubusercontent.com/86934170/213891970-99d76854-9556-41d7-be20-519e37b63519.mp4

**Comment by bdach** [comment]
Then we have nothing to go on. This is where the only advice left is to update drivers or something, because I have no idea how to begin diagnosing this with zero error details and no reproduction (doesn't happen to me).

**Comment by Loreos7** [comment]
There is nothing wrong on my side. My drivers are up-to-date.
Let's start from this video then (this is the previous release):

https://user-images.githubusercontent.com/86934170/213894832-e86e7b3c-eef6-4ca8-a9f1-d33f4a95c87f.mp4

**Comment by bdach** [comment]
I'm converting to discussion since the issue is non-specific and likely hardware-dependent.

**Comment by Joppe27** [comment, marked as answer]
I can reproduce this. When testing, make sure to enter gameplay *without* autoplay, because otherwise this application hang will not occur.

Taking a very quick look into what's causing this, it seems like the update thread gets stuck inside of this `while` loop:
https://github.com/ppy/osu/blob/7d1b5751fdb4428da009e6cb9ffa2ef143723a6e/osu.Game/Database/RealmArchiveModelImporter.cs#L569-L573

Which in turn is caused by the skin copy/import below being put on hold for the entirety of the gameplay duration, just like any other skin during import (to avoid impacting gameplay performance, but in this specific case that's a problem): 
https://github.com/ppy/osu/blob/7d1b5751fdb4428da009e6cb9ffa2ef143723a6e/osu.Game/Skinning/SkinManager.cs#L172-L180

Another video of the application hang:

https://user-images.githubusercontent.com/56885706/215738678-354ec66f-68ec-422c-a7af-fe9b6d5f46f6.mp4

**Reply by bdach** [reply]
Thanks, this is actually helpful. The original report did not bother to use enough words I don't think.

Tracking as https://github.com/ppy/osu/issues/22486.
